src/main/resources/static/api.yml

```diff
@@ -1756,3 +1787,30 @@ components:
           format: date-time
       required:
         - data
+    BehaviourNote:
+      type: object
+      properties:
+        id:
+          type: long
+        createdAt:
+          type: string
+          format: date-time
+        occurredAt:
+          type: string
+          format: date-time
+        authorName:
+          type: string
+        type:
+          type: string
+        subType:
+          type: string
+        note:
+          type: string
+      required:
+        - id
+        - createdAt
+        - occurredAt
+        - authorName
+        - type
```

Comment:
I did consider this but went with plain strings because the subTypes are different per type and as far as I know OpenAPI doesn't have a way to represent that kind of relationship so this is potentially more confusing than not having them listed at all.  Aditionally Ed thinks we should go for an exclude-categories list rather than an include, so listing them all now increases the chances of a new category being added to NOMIS breaking our system (i.e. we'd need to do a release and add a new entry to the enum each time a category was added to NOMIS.)

That's my thoughts anyway, happy to put these in if you'd prefer?

Reply:
just to add a bit of background, the potentially useful case notes which we've encountered (just looking at a handful of people) includes the following types:

- Report | Residence
- General | Resettlement
- Key Worker Activity
- General | History
- Communication | Residence
- Accredited Programme | Deselection
- Negative Behaviour | General Entry
- Negative Behaviour | Behaviour Warning
- Positive behaviour | General Entry
- General | Well--Being check
- Social care | Social care
- General | Offender supervisor entry

As we don't know in advance which are the case note types which are of value, I'm suggesting that we make it easy to progressively exclude those types/sub-types which, during testing, we identify as being irrelevant. 


See https://trello.com/c/TNxixi2l/757 for more information